src/Symfony/Bundle/SecurityBundle/Security/Http/RememberMe/RememberMeServicesResolver.php

```diff
@@ -0,0 +1,52 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\SecurityBundle\Security\Http\RememberMe;
+
+use Symfony\Component\DependencyInjection\ContainerInterface;
+use Symfony\Component\Security\Http\RememberMe\RememberMeServicesInterface;
+use Symfony\Component\Security\Http\RememberMe\RememberMeServicesResolverInterface;
+
+class RememberMeServicesResolver implements RememberMeServicesResolverInterface
+{
+    /**
+     * @var \Symfony\Component\DependencyInjection\ContainerInterface
+     */
+    private $serviceContainer;
+
+    /**
+     * @param ContainerInterface $serviceContainer
+     */
+    public function __construct(ContainerInterface $serviceContainer)
+    {
+        $this->serviceContainer = $serviceContainer;
+    }
+
+    /**
+     * @param $providerKey
+     * @return null|RememberMeServicesInterface
+     */
+    public function resolve($providerKey)
+    {
+        $rememberMeServices = null;
+        if ($this->serviceContainer->has('security.authentication.rememberme.services.persistent.'.$providerKey)) {
```

Comment:
Do you need them to be public? Or put in other words: what is your reason to use the container here instead of using a compiler pass (executed before the pass that removes definitions) as suggested by @iltar in https://github.com/symfony/symfony/pull/13062/files#r22148006?